src/test/java/com/restapi/carMarket/service/CarServiceUnitTest.java

```diff
@@ -11,168 +8,199 @@
-        carService.deleteById(id);
+        carService.removeCarFromMarketById(id);
 
         Mockito.verify(carDAO,Mockito.times(1)).deleteById(id);
     }
 
     @Test(expected = CarNotFoundException.class)
-    public void deleteNonExistentCarByIdShouldThrowCarNotFoundException() throws Exception{
+    public void deleteNonExistentCarByIdShouldThrowCarNotFoundException() {
         Long id = 10L;
 
         given(carDAO.existsById(id)).willReturn(false);
-        carService.deleteById(id);
+        carService.removeCarFromMarketById(id);
 
         Mockito.verify(carDAO,Mockito.times(1)).deleteById(id);
     }
 
     @Test
-    public void deleteExistentCarShouldCallDeleteMethodOfDAO() throws Exception{
+    public void deleteExistentCarShouldCallDeleteMethodOfDAO() {
         Car car = new Car("BMW","5 Series", 2010, 40000);
 
         given(carDAO.exists(Example.of(car))).willReturn(true);
-        carService.delete(car);
+        carService.removeCarFromMarket(car);
 
         Mockito.verify(carDAO,Mockito.times(1)).delete(car);
     }
 
     @Test(expected = CarNotFoundException.class)
-    public void deleteNonExistentCarShouldThrowCarNotFoundException() throws Exception{
+    public void deleteNonExistentCarShouldThrowCarNotFoundException() {
         Car car = new Car("BMW","5 Series", 2010, 40000);
 
         given(carDAO.exists(Example.of(car))).willReturn(false);
-        carService.delete(car);
+        carService.removeCarFromMarket(car);
 
         Mockito.verify(carDAO,Mockito.times(0)).delete(car);
     }
 
     @Test
-    public void updateExistentCarWithValidCarShouldUpdateTheObject() throws Exception{
+    public void updateExistentCarWithValidCarShouldUpdateTheObject() {
         Long id = 10L;
         Car newCar = new Car("BMW","7 Series", 2010, 10000);
         Car oldCar = new Car("BMW","5 Series", 2010, 10000);
 
         given(carDAO.findById(id)).willReturn(Optional.of(oldCar));
-        carService.update(id, newCar);
+        carService.updateCarInformation(id, newCar);
 
         Assert.assertTrue(newCar.equals(oldCar));
     }
 
-    @Test(expected = CarNotFoundException.class)
-    public void updateNonExistentCarWithValidCarShouldThrowCarNotFoundException() throws Exception{
+    @Test
+    public void updateNonExistentCarWithValidCarShouldSaveNewCar() {
         Long id = 10L;
         Car newCar = new Car("BMW","7 Series", 2010, 10000);
 
-        carService.update(id, newCar);
+        given(carDAO.findById(id)).willReturn(Optional.empty());
+
+        carService.updateCarInformation(id, newCar);
+
+        Mockito.verify(carDAO,Mockito.times(1)).save(any(Car.class));
     }
 
     @Test(expected = CarNotValidException.class)
     public void updateExistentCarWithInValidCarShouldThrowCarNotValidException(){
         Long id = 10L;
         Car newCar = new Car("BMW","7 Series", 0, 10000);
 
-        carService.update(id, newCar);
+        carService.updateCarInformation(id, newCar);
```

Comment:
We test that a new car is valid, but what if the car exists by the new car values are not valid? For example, the car saved in the database had model but the new car has null for model? I don't see tests for that.

Reply:
Sorry, I just used one test case of invalid car ( car with invalid price) to test the exception throw. I will add test cases for invalidiness of each attribute.

Reply:
done